File: Monopoly/Monopoly/Railroads.java
package Monopoly;

public class Railroads extends Property{

    int rent;

    public Railroads(int location, Group group, int price) {
        super(location, group, price);
    }

    @Override
    public boolean CanBeMortaged() {
        return false;
    }

    @Override
    public int GetRent() {
        return rent;
    }

    @Override
    public int GetAction() {
        return 0;
    }
}

File: PA3UI/PA3BackEnd/src/Monopoly/Railroads.cs
using System;

namespace PA3BackEnd.src.Monopoly
{
    public class Railroads: Property
    {
        public Railroads(int location, Group group, int price, string name) : base(location, group, price, name) {
            
        }
        
        public override bool CanBeMortaged() {
            return !isMortaged;
        }

        public override bool CanPlayerBuild(int playerId)
        {
            return false;
        }

        public override int GetRent() {
            if (isMortaged)
            {
                return 0;
            }
            int amount = this.group.GetAmountPlayerOwns(owner);
            return ((int)Math.Pow(2, amount-1))*25;
        }
    }
}